index.bs

```diff
@@ -692,44 +692,75 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. For each |authenticator| currently available on this platform, perform the following steps:
 
-    <span class="note">Note: the procedure to determine the [=authenticators=] present on a platform is 
-        platform-specific.</span>
-
-    1. Let |credentialList| be a new [=ordered set=].
+    1. Let |credentialDescriptorList| be a new [=ordered set=] of {{ScopedCredentialDescriptor}}.
 
     1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
         <dl class="switch">
             : [=list/is not empty=]
-            :: [=set/For each=] credential |C| in <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
-                1. If there is any credential |X| present on this |authenticator| that:
+            :: [=set/For each=] {{ScopedCredentialDescriptor}} |C| in
+                <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+                1. For each credential |X| present on this |authenticator| that:
                     - is mapped to |rpId|, and,
                     - matches with <code>|C|.{{ScopedCredentialDescriptor/id}}</code> and 
                         <code>|C|.{{ScopedCredentialDescriptor/type}}</code>,
 
-                    then [=set/append=] |X| to |credentialList|.
+                    then:
+                        1. let |Y| be a new {{ScopedCredentialDescriptor}}.
```

Comment:
It seems like you could just append |C| to |credentialDescriptorList| here. You could copy it if you're worried about modifications later, but dictionaries are already copied on their way into IDL algorithms. And |Y| differs from |C| in its value for `transports`, but it seems like you'd rather use |C|'s.